### PR TITLE
396 markdown image upload

### PR DESCRIPTION
**Pull request by jcalfee**
Configure to use the imagehoster via: `"uploadImage": "https://..."`

### CODE REVIEW

**Comment by sneak** [comment]
I didn't see it in this PR - but if the client key-based auth to the imagehoster is signing the challenge and not a hash of the challenge, the challenge needs to be prefixed with a constant to make sure the server can't easily make the client sign a transaction doing something else. @jcalfee

**Comment by jcalfee** [comment]
Prefix challenge is in production and stable...

File: app/redux/UserSaga.js [hunk 1]
```diff
@@ -429,25 +431,31 @@ function* uploadImage({payload: {file, dataUrl, filename = 'image.txt', progress
     const sig = Signature.signBufferSha256(bufSha, d)
     const postUrl = `${$STM_Config.uploadImage}/${username}/${sig.toHex()}`
```
Comment (sneak) [inline app/redux/UserSaga.js]: all the other keys in $STM_Config are snake_case and not camelCase.

**Comment by roadscape** [comment]
Tested working, just needs some server-side config.

**Pull request merged**